src/main/java/blusunrize/immersiveengineering/api/excavator/ExcavatorHandler.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * BluSunrize
+ * Copyright (c) 2020
+ *
+ * This code is licensed under "Blu's License of Common Sense"
+ * Details can be found in the license file in the root folder of this project
+ *
+ */
+
+package blusunrize.immersiveengineering.api.excavator;
+
+import blusunrize.immersiveengineering.common.IESaveData;
+import blusunrize.immersiveengineering.common.util.Utils;
+import com.google.common.collect.ArrayListMultimap;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.util.math.ColumnPos;
+import net.minecraft.world.World;
+import net.minecraft.world.dimension.DimensionType;
+import net.minecraft.world.gen.INoiseGenerator;
+import org.apache.commons.lang3.tuple.Pair;
+
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * @author BluSunrize - 03.06.2015
+ * <p>
+ * The Handler for the Excavator. Generates veins upon world gen, stores them, and keeps a cache of queried positions
+ */
+public class ExcavatorHandler
+{
+	private static final ArrayListMultimap<DimensionType, MineralVein> MINERAL_VEIN_LIST = ArrayListMultimap.create();
+	private static final HashMap<ColumnPos, MineralWorldInfo> MINERAL_INFO_CACHE = new HashMap<>();
```

Comment:
I don't remember whether we've talked about this before, but I prefer to use the most "abstract" type I can get away with for variables (within reason, so not `Collection<T>` over `Set<T>` in most cases). So this should be a `Multimap` and a `Map`, not an `ArrayListMultimap` and a `HashMap`.

Edit: Same for the accessor methods. Even more relevant there, in case we decide to switch to a slightly different type later on.